tests/validation.py

```diff
@@ -0,0 +1,36 @@
+import pytest
+
+from ldrshared.validation import ldrschemas, ValidationError
+
+
+def test_confirm_indexer_trigger_message_allows_valid():
+    """
+    Check that confirm_indexer_trigger_message allows a valid message schema
+    """
+    valid_message = {"input_data_path": "foo"}
+    ldrschemas.confirm_indexer_trigger_message(valid_message)
+
+
+def test_confirm_indexer_trigger_message_raises_with_missing_key():
+    """
+    Check that confirm_indexer_trigger_message raise for a missing
+    required key
+    """
+
+    missing_key_message = {"wrong_key": "foo"}
+    with pytest.raises(ValidationError):
+        ldrschemas.confirm_indexer_trigger_message(missing_key_message)
+
+
+def test_confirm_indexer_trigger_message_raises_with_too_many_keys():
```

Comment:
Do we have any CI set up to run these on commit? If not could you raise a ticket for some?

Reply:
I'll just add it, it's not hard, off topic but we should probably have repo template for apps that includes a test on pr action as default.

This is even more off topic, but since actions call other github actions, we might benefit from defining **a** ci/cd action that just gets included in every SE repo.

- always run pytest (where present) and behave (where present) on pr.
- run black and reject a pr if that creates a diff.
- reject if there's any `__pycache__` type files in there.
- if there's a develop branch push to dockerhub as `reponame:develop` (might want to toggle this with config, not everything gets a container)
- if there's a master branch push to dockerhub as `reponame:master` (might want to toggle this with config, not everything gets a container)

That's off the cuff, but you could go pretty deep and centrally controlled it'd be pretty nice. Every repo just needs a single action (ours) then magic happens.